proto/opamp.proto

```diff
@@ -817,3 +833,28 @@ message AgentConfigFile {
     // example "text/yaml".
     string content_type = 2;
 }
+
+/////////////////////////////////////////////////////////////////////////////////////
+// Custom messages
+/////////////////////////////////////////////////////////////////////////////////////
+
+message CustomCapabilities {
+    // A list of custom capabilities that are supported. Each capability is a reverse
+    // FQDN that uniquely identifies the custom capability.
+    // Status: [Development]
+    repeated string capabilities = 1;
+}
+
+message CustomMessage {
+    // A reverse FQDN that uniquely identifies the custom capability.
+    // Status: [Development]
+    string capability = 1;
+
+    // The custom message type.
+    // Status: [Development]
+    string type = 2;
```

Comment:
I may be missing context on this but would it make sense to just combine capability and type into one field? It seems like `capability.type` is just as useful and a little simpler.

Reply:
I wanted the capability to match the one in the CustomCapabilities message.

Reply:
I see, a particular capability is associated with a set of message types. Those types are _implied_ rather than declared.